packages/types/src/connection/pool.ts

```diff
@@ -0,0 +1,28 @@
+export interface ConnectionPool<T> {
```

Comment:
for this and ConnectionManager, are there any practical limits on what `T` is that should be expressed with `T extends ...`?

Reply:
So we are only implementing this interface for http2 and `T` is essentially `ClientHttp2Session`.

The Node.js http module manages connections internally using httpAgent, and browser does not expose any APIs for managing connections. So the `ConnectionPool` and `ConnectionManager` interfaces might not need generics. 

I will confirm with @trivikr about this.